distributed/tests/test_ssl.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import print_function, division, absolute_import
+
+import subprocess
+
+from distributed import (Client, Scheduler, Worker)
+from distributed.utils_test import gen_cluster, inc, loop
+
+import pytest
+import os
+
+@pytest.fixture(scope="function")
+def ssl_kwargs(tmpdir):
+
+    ssl_dir = tmpdir.mkdir("ssl")
+    dirname = ssl_dir.dirname
+
+    keyfile = os.path.join(dirname, 'test.key')
+    certfile = os.path.join(dirname, 'test.pem')
+
+    p = subprocess.Popen([
+        'openssl', 'req',
+        '-x509',
+        '-nodes',
+        '-newkey',
+        'rsa:2048',
+        '-days', '365',
+        '-subj', "/O=dask/CN=localhost"
+        '-nodes',
+        '-passout', 'pass:',
+        '-keyout', 'test.key',
+        '-out', 'test.pem'],
+        cwd=ssl_dir.dirname,
+    )
+
+    p.communicate()
+    assert p.returncode == 0
+
+    return {'ssl_options': {
```

Comment:
I'd rather make it mandatory for it to be a SSLContext. I understand that Tornado accepts several different things for compatibility reasons, but we are not tied by backwards compatibility here.